File: app/src/main/java/com/example/sportstracker/di/components/AppComponent.java
package com.example.sportstracker.di.components;

import android.app.Application;

import com.example.sportstracker.SportsTrackerApp;
import com.example.sportstracker.di.module.ActivityModule;
import com.example.sportstracker.di.module.AppModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

/**
 * The main application component which initializes all the dependent modules
 * Author: Lajesh D
 * Email: lajeshds2007@gmail.com
 * Created: 7/24/2018
 * Modified: 7/24/2018
 */
@Singleton
@Component(modules = {
        AppModule.class,
        AndroidInjectionModule.class,
        ActivityModule.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

    void inject(SportsTrackerApp sportsTrackerApp);
}

File: app/src/main/java/com/example/sportstracker/di/components/AppComponent.kt
package com.example.sportstracker.di.components

import android.app.Application
import com.example.sportstracker.SportsTrackerApp
import com.example.sportstracker.di.module.ActivityModule
import com.example.sportstracker.di.module.AppModule
import com.example.sportstracker.di.module.NetworkModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, ActivityModule::class, NetworkModule::class, AndroidInjectionModule::class])
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application?): Builder
        fun build(): AppComponent
    }

    fun inject(sportsTrackerApp: SportsTrackerApp)
}